pkg/storage/segments/repairer.go

```diff
@@ -29,18 +30,37 @@ type Repairer struct {
 	ec       ecclient.Client
 	identity *identity.FullIdentity
 	timeout  time.Duration
+
+	// multiplierOptimalThreshold is the value that multiplied by the optimal
+	// threshold results in the maximum limit of number of nodes to upload
+	// repaired pieces
+	multiplierOptimalThreshold float64
 }
 
-// NewSegmentRepairer creates a new instance of SegmentRepairer
-func NewSegmentRepairer(log *zap.Logger, metainfo *metainfo.Service, orders *orders.Service, cache *overlay.Cache, ec ecclient.Client, identity *identity.FullIdentity, timeout time.Duration) *Repairer {
+// NewSegmentRepairer creates a new instance of SegmentRepairer.
+//
+// excessPercentageOptimalThreshold is the percentage to apply over the optimal
+// threshould to determine the maximum limit of nodes to upload repaired pieces,
+// when negative, 0 is applied.
+func NewSegmentRepairer(
+	log *zap.Logger, metainfo *metainfo.Service, orders *orders.Service,
+	cache *overlay.Cache, ec ecclient.Client, identity *identity.FullIdentity, timeout time.Duration,
+	excessPercentageOptimaThreshold int,
+) *Repairer {
+
+	if excessPercentageOptimaThreshold < 0 {
+		excessPercentageOptimaThreshold = 0
+	}
+
 	return &Repairer{
-		log:      log,
-		metainfo: metainfo,
-		orders:   orders,
-		cache:    cache,
-		ec:       ec.WithForceErrorDetection(true),
-		identity: identity,
-		timeout:  timeout,
+		log:                        log,
+		metainfo:                   metainfo,
+		orders:                     orders,
+		cache:                      cache,
+		ec:                         ec.WithForceErrorDetection(true),
+		identity:                   identity,
+		timeout:                    timeout,
+		multiplierOptimalThreshold: 1 + float64((excessPercentageOptimaThreshold / 100)),
```

Comment:
um I think `float64((excessPercentageOptimalThreshold / 100))` will do integer division before the cast to a `float64`. In other words, I think it will always be 0 unless `excessPercentageOptimalThreshold` is 100. I think you need to do `float64(excessPercentageOptimalThreshold) / 100` or something. See: https://play.golang.org/p/KX0cKErSsbE

Reply:
Because the configuration value has been changed from int to float64, this bug doesn't exist anymore